sway-core/src/asm_generation/miden_vm/miden_vm_asm_builder.rs

```diff
@@ -679,3 +683,22 @@ impl ToMidenBytecode for Vec<DirectOp> {
             .join("\n")
     }
 }
+
+struct MidenFinalProgram {
+    ops: Vec<crate::asm_generation::DirectOp>,
+}
+
+impl MidenFinalProgram {
+    fn finalize(self) -> FinalizedAsm {
+        FinalizedAsm {
+            data_section: DataSection {
+                ..Default::default()
+            },
+            program_section: InstructionSet::MidenVM { ops: self.ops },
+            // should this be a script? :think:
+            program_kind: ProgramKind::Script,
+            entries: vec![],
+            abi: None, /* TODO? */
```

Comment:
What's TODO?

Reply:
Not sure... This was already in the code: https://github.com/FuelLabs/sway/blob/master/sway-core/src/asm_generation/programs/final.rs

Reply:
I guess we can leave it for the miden guys to figure out if they get back to implementing this.

Reply:
I guess we'll leave it to the miden guys to figure out when they get back to implementing this.